addons/account/tests/test_account_move_entry.py

```diff
@@ -677,3 +677,32 @@ def test_misc_prevent_edit_tax_on_posted_moves(self):
         # You can remove journal items if the related journal entry is draft.
         self.test_move.button_draft()
         self.assertTrue(self.test_move.line_ids.unlink())
+
+    def test_draft_name_on_move_to_period(self):
+        # When a draft entry is added to an empty period, it should get a name.
+        # When a draft entry with a name is moved to a period already having entries, its name should be reset to '/'.
+        new_move = self.env['account.move'].create({
+            'move_type': 'entry',
+            'date': fields.Date.from_string('2016-02-01'),
+            'line_ids': [
+                (0, None, {
+                    'name': 'revenue line 1',
+                    'account_id': self.company_data['default_account_revenue'].id,
+                    'debit': 500.0,
+                    'credit': 0.0,
+                }),
+                (0, None, {
+                    'name': 'revenue line 2',
+                    'account_id': self.company_data['default_account_revenue'].id,
+                    'debit': 0.0,
+                    'credit': 500.0,
+                }),
+            ]
+        })
+
+        # Empty period, so a name should be set
+        self.assertNotEqual(new_move.name, '/')
+        # Move to an existing period with another move in it
+        new_move.date = fields.Date.from_string('2016-01-10')
+        # Not an empty period, so name should be rest to '/' (draft)
+        self.assertEqual(new_move.name, '/')
```

Comment:
What happens if you switch back to `2016-02-01` now?

Reply:
For one move, it works fine and a sequence is assigned. When working with multiple records, all but the first are set to `/`. Will correct and extend the current test.